domain-management/src/main/java/org/jboss/as/domain/management/security/FileKeyManagerService.java

```diff
@@ -117,45 +141,117 @@ public void setAlias(String alias) {
-        }
+            }
 
-        keyStore = FileKeystore.newKeyStore(provider, file, keystorePassword, keyPassword, alias);
-        keyStore.load();
+            keyStore = FileKeystore.newKeyStore(provider, file, keystorePassword, keyPassword, alias);
+            keyStore.load();
 
-        super.start(context);
+            return keyStore.getKeyStore();
+        } catch (StartException e) {
+            throw new IllegalStateException(e);
+        }
     }
 
-    @Override
-    public void stop(StopContext context) {
-        super.stop(context);
-        keyStore = null;
+    private void generateFileKeyStore(Path path) {
+        try {
+            DomainManagementLogger.SECURITY_LOGGER.keystoreWillBeCreated(path.toString());
+            KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+            keyGen.initialize(2048, new SecureRandom());
+            KeyPair pair = keyGen.generateKeyPair();
+            X509Certificate cert = generateCertificate(pair);
+
+
+            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keyStore.load(null, keystorePassword);
+
+            //Generate self signed certificate
+            X509Certificate[] chain = new X509Certificate[1];
+            chain[0] = cert;
+            keyStore.setKeyEntry(alias, pair.getPrivate(), keyPassword, chain);
+            try (FileOutputStream stream = new FileOutputStream(path.toFile())) {
+                keyStore.store(stream, keystorePassword);
+            }
+
+        } catch (Exception e) {
+            throw DomainManagementLogger.SECURITY_LOGGER.failedToGenerateSelfSignedCertificate(e);
+        }
     }
 
-    @Override
-    protected KeyStore loadKeyStore() {
-        return keyStore.getKeyStore();
+    X509Certificate generateCertificate(KeyPair pair) throws Exception {
+        PrivateKey privkey = pair.getPrivate();
+        X509CertInfo info = new X509CertInfo();
+        Date from = new Date();
+        Date to = new Date(from.getTime() + (1000L * 60L * 60L * 24L * 365L * 10L));
+        CertificateValidity interval = new CertificateValidity(from, to);
+        BigInteger sn = new BigInteger(64, new SecureRandom());
+        X500Name owner = new X500Name("CN=" + autoGenerateCertHostName);
+
+        info.set(X509CertInfo.VALIDITY, interval);
+        info.set(X509CertInfo.SERIAL_NUMBER, new CertificateSerialNumber(sn));
+        info.set(X509CertInfo.SUBJECT, owner);
+        info.set(X509CertInfo.ISSUER, owner);
+        info.set(X509CertInfo.KEY, new CertificateX509Key(pair.getPublic()));
+        info.set(X509CertInfo.VERSION, new CertificateVersion(CertificateVersion.V3));
+        AlgorithmId algo = new AlgorithmId(AlgorithmId.sha256WithRSAEncryption_oid);
+        info.set(X509CertInfo.ALGORITHM_ID, new CertificateAlgorithmId(algo));
+
+        // Sign the cert to identify the algorithm that's used.
+        X509CertImpl cert = new X509CertImpl(info);
```

Comment:
It'd be nice to use the Elytron DER-encoding API instead of (possibly JDK-specific) internal classes to build up the certificate data.  Maybe we should introduce a CertificateBuilder in fact...